Makefile

```diff
@@ -67,24 +67,23 @@ build-tf2-gpu:
 
 .PHONY: publish-tf1-cpu
 publish-tf1-cpu:
-	docker push $(CPU_TF1_ENVIRONMENT_NAME)-$(SHORT_GIT_HASH)
-	docker push $(CPU_TF1_ENVIRONMENT_NAME)-$(VERSION)
+	scripts/publish-docker.sh tf1-cpu $(CPU_TF1_ENVIRONMENT_NAME) $(SHORT_GIT_HASH) $(VERSION)
 
 .PHONY: publish-tf2-cpu
 publish-tf2-cpu:
-	docker push $(CPU_TF2_ENVIRONMENT_NAME)-$(SHORT_GIT_HASH)
-	docker push $(CPU_TF2_ENVIRONMENT_NAME)-$(VERSION)
+	scripts/publish-docker.sh tf2-cpu $(CPU_TF2_ENVIRONMENT_NAME) $(SHORT_GIT_HASH) $(VERSION)
 
 .PHONY: publish-tf1-gpu
 publish-tf1-gpu:
-	docker push $(GPU_TF1_ENVIRONMENT_NAME)-$(SHORT_GIT_HASH)
-	docker push $(GPU_TF1_ENVIRONMENT_NAME)-$(VERSION)
+	scripts/publish-docker.sh tf1-gpu $(GPU_TF1_ENVIRONMENT_NAME) $(SHORT_GIT_HASH) $(VERSION)
 
 .PHONY: publish-tf2-gpu
 publish-tf2-gpu:
-	docker push $(GPU_TF2_ENVIRONMENT_NAME)-$(SHORT_GIT_HASH)
-	docker push $(GPU_TF2_ENVIRONMENT_NAME)-$(VERSION)
+	scripts/publish-docker.sh tf2-gpu $(GPU_TF2_ENVIRONMENT_NAME) $(SHORT_GIT_HASH) $(VERSION)
 
 .PHONY: publish-cloud-images
 publish-cloud-images:
-	cd cloud && packer build -var "image_suffix=-$(SHORT_GIT_HASH)" environments-packer.json
+	mkdir -p /tmp/artifacts
```

Comment:
non-blocking: Maybe make the artifact path a variable at the top of this file and an argument to `publish-docker.sh` for DRY's sake.

Reply:
sure